src/testing/euler_split_inviscid_taylor_green_vortex.cpp

```diff
@@ -11,236 +11,193 @@ EulerTaylorGreen<dim, nstate>::EulerTaylorGreen(const Parameters::AllParameters
-//
-//        AssertDimension(all_parameters.dimension, PHILIP_DIM);
-//
-//        std::cout << "Starting program..." << std::endl;
-//
-//  using namespace PHiLiP;
-//  //const Parameters::AllParameters parameters_input;
-//  EulerTaylorGreen<PHILIP_DIM, PHILIP_DIM+2> euler_test(&all_parameters);
-//  int i = euler_test.run_test();
-//  return i;
-// }
-// catch (std::exception &exc)
-// {
-//  std::cerr << std::endl << std::endl
-//      << "----------------------------------------------------"
-//      << std::endl
-//      << "Exception on processing: " << std::endl
-//              << exc.what() << std::endl
-//              << "Aborting!" << std::endl
-//              << "----------------------------------------------------"
-//              << std::endl;
-//  return 1;
-// }
-//
-// catch (...)
-// {
-//     std::cerr << std::endl
-//               << std::endl
-//               << "----------------------------------------------------"
-//               << std::endl
-//               << "Unknown exception!" << std::endl
-//               << "Aborting!" << std::endl
-//               << "----------------------------------------------------"
-//               << std::endl;
-//     return 1;
-// }
-// std::cout << "End of program." << std::endl;
-// return test_error;
-//}
+    std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation> (mpi_communicator);
+
+    PHiLiP::Parameters::AllParameters all_parameters_new = *all_parameters;  
+    double left = 0.0;
+    double right = 2 * dealii::numbers::PI;
+    const bool colorize = true;
+    int n_refinements = 2;
+    unsigned int poly_degree = 3;
+    const unsigned int grid_degree = 1;
+    dealii::GridGenerator::hyper_cube(*grid, left, right, colorize);
+     
+    std::vector<dealii::GridTools::PeriodicFacePair<typename dealii::Triangulation<PHILIP_DIM>::cell_iterator> > matched_pairs;
+    dealii::GridTools::collect_periodic_faces(*grid,0,1,0,matched_pairs);
+    dealii::GridTools::collect_periodic_faces(*grid,2,3,1,matched_pairs);
+    dealii::GridTools::collect_periodic_faces(*grid,4,5,2,matched_pairs);
+    grid->add_periodicity(matched_pairs);
+     
+    grid->refine_global(n_refinements);
+
+    const unsigned int n_global_active_cells2 = grid->n_global_active_cells();
+    double n_dofs_cfl = pow(n_global_active_cells2,dim) * pow(poly_degree+1.0, dim);
+    double delta_x = (right-left)/pow(n_dofs_cfl,(1.0/dim)); 
+    all_parameters_new.ode_solver_param.initial_time_step =  0.1 * delta_x;
+    pcout<<" timestep "<<all_parameters_new.ode_solver_param.initial_time_step<<std::endl;
+
+    //Create DG
+    std::shared_ptr < PHiLiP::DGBase<dim, double> > dg = PHiLiP::DGFactory<dim,double>::create_discontinuous_galerkin(&all_parameters_new, poly_degree, poly_degree, grid_degree, grid);
+    dg->allocate_system ();
+
+    std::cout << "Implement initial conditions" << std::endl;
+    dealii::FunctionParser<PHILIP_DIM> initial_condition(5);
```

Comment:
Since there is a templated parameter `dim` available, `dealii::FunctionParser<PHILIP_DIM> initial_condition(5)` can be replaced by `dealii::FunctionParser<dim> initial_condition(5)`?

Reply:
fixed